### PR TITLE
Extend new annotations to SortingInterfaces

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

Extending the new `FilePathType` and `FolderPathType` usage in our localized method signatures for the `SortingInterfaces`.

@luiztauffer If you could check these similar to the previous PR, see if they work OK for you.

## How to test the behavior?
```
pytest
```

The existing tests verify the schema validity for all these changes.

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?

### CODE REVIEW

File: nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py [hunk 1]
```diff
@@ -35,7 +30,6 @@ def __init__(self, **source_data):
         super().__init__(**source_data)
         self.recording_extractor = self.RX(**source_data)
         self.subset_channels = None
-        self.source_data = source_data
```
Comment (CodyCBakerPhD) [inline nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py:38]: Cleanup - `self.source_data` is set during the `super().__init__` 3 lines above

File: nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py [hunk 2]
```diff
@@ -7,15 +7,10 @@
...
+from pynwb.ecephys import ElectrodeGroup
 
 from ...basedatainterface import BaseDataInterface
-from ...utils.json_schema import (
-    get_schema_from_hdmf_class,
-    get_schema_from_method_signature,
-    fill_defaults,
-    get_base_schema,
-)
+from ...utils.json_schema import get_schema_from_hdmf_class, get_schema_from_method_signature, get_base_schema
```
Comment (CodyCBakerPhD) [inline nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py:13]: import cleanup

File: nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py [hunk 3]
```diff
@@ -24,7 +19,7 @@ class BaseSortingExtractorInterface(BaseDataInterface, ABC):
     @classmethod
     def get_source_schema(cls):
         """Compile input schema for the SortingExtractor."""
-        return get_schema_from_method_signature(cls.SX.__init__)
+        return get_schema_from_method_signature(cls.__init__)
```
Comment (CodyCBakerPhD) [inline nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py:22]: The actual biggest change here, even though it's only 1 line. Changes all the method signatures from having to rely on spikeextractors to our own local interfaces for them

**Review by bendichter — APPROVED** [review]
looks great!

**Pull request merged**